allennlp/nn/util.py

```diff
@@ -344,23 +342,24 @@ def sequence_cross_entropy_with_logits(logits: torch.FloatTensor,
     if batch_average:
         num_non_empty_sequences = ((weights.sum(1) > 0).float().sum() + 1e-13)
         return per_batch_loss.sum() / num_non_empty_sequences
-    return per_batch_loss.squeeze(1)
+    return per_batch_loss
 
 
 def replace_masked_values(tensor: Variable, mask: Variable, replace_with: float) -> Variable:
     """
     Replaces all masked values in ``tensor`` with ``replace_with``.  ``mask`` must be broadcastable
     to the same shape as ``tensor``.
-
-    Until pytorch 0.2 is released, we require that ``tensor.dim() == mask.dim()``.  Otherwise we
-    won't know which dimensions of the mask to unsqueeze.
     """
-    # We'll build a tensor of the same shape as `tensor`, subtract away masked values, then add
-    # back in the `replace_with` value.
-    # TODO(mattg): use broadcasting here when it's available in pytorch 0.2.
-    if tensor.dim() != mask.dim():
-        raise ConfigurationError("tensor.dim() (%d) != mask.dim() (%d)" % (tensor.dim(), mask.dim()))
-    one_minus_mask = 1.0 - mask.expand_as(tensor)
+    # We'll build a tensor of the same shape as `tensor`, subtract away
+    # masked values, then add back in the `replace_with` value.
+    if tensor.dim() < mask.dim():
+        raise ConfigurationError("Mask has dimension {} but tensor has dimension {}. Mask dimension "
+                                 "should be <= tensor dimension.".format(mask.dim(), tensor.dim()))
+    # Expand mask dimensions to match tensor if it's smaller.
+    while mask.dim() < tensor.dim():
+        mask = mask.unsqueeze(-1)
```

Comment:
Yeah ... I wasn't quite sure when doing this that this was what you meant - i'll reverse it. The bidaf tests are still passing because I didn't remove the `mask.unsqueeze(-1)` before the function is called, making the mask the same shape as the tensor anyway.